.github/workflows/flowzone.yml

```
@@ -1334,6 +1334,38 @@ jobs:
           json: ${{ toJSON(inputs) }}
           secrets: ${{ toJSON(secrets) }}
 
+  custom_always:
+    name: Always custom
+    runs-on: ${{ fromJSON(inputs.runs_on) }}
+    timeout-minutes: ${{ fromJSON(inputs.jobs_timeout_minutes) }}
+    needs:
+      - event_types
+      - project_types
+      - versioned_source
+      - custom_test
+      - custom_publish
+      - custom_finalize
+      - custom_clean
+    if: always()
```

I don't think this should run if event_types is skipped or failed?

---

@klutchell I've just made it dependent on any of the other `test_*` events. I think that should be safe.

---

This job needs to depend on `needs.project_types.outputs.custom_always == 'true'` and I still think we need to depend on `needs.event_types.result == 'success' to make sure we don't run if the event type cannot be parsed or is skipped.

---

The `custom_always` job should not run if `actions/custom/action.yml` is not found, it will fail. Like I said, try this on a repo where that file does not exist and you'll see what I mean.

---

so basically changing from `always()` to this, right?

```
...
  custom_always:
    name: Always custom
    runs-on: ${{ fromJSON(inputs.runs_on) }}
    timeout-minutes: ${{ fromJSON(inputs.jobs_timeout_minutes) }}
    needs:
      - event_types
      - custom_test
      - custom_publish
      - custom_finalize
      - custom_clean
    if: needs.project_types.outputs.custom_always == 'true'

    steps:
...
```

---

```suggestion
    if: !failure() &&
      needs.event_types.result == 'success' &&
      needs.project_types.outputs.custom_always == 'true'
```

This would always run unless a previous job failed. If you still want to run in the case of failures change `!failure()` back to `always()` but keep the other two checks.